### ISSUE TITLE
[HELP] How do you uninstall Go?

### ISSUE THREAD

**Issue by obvionaoe** [darkwizard242/ansible-role-go#5]
How do you use this role to uninstall Go?

Thanks

**Comment by darkwizard242** [comment]
Hi @obvionaoe - Unlike installation of packages using native package managers such as apt/yum/dnf/pacman etc , Go, unfortunately isn't installed on Linux via package managers.

The role at present is missing the feature of uninstallation - which I am glad you brought. up. I will be adding a feature to uninstall go via the role sometime this week.

However, there is still a work around if you had installed go using this role. Simply put you will only need to remove a folder and  a file via commandline.

Assuming  that you didn't customize the value for `go_bin_path` var. You can use either of the following methods to uninstall:

#### If as `root`:
```shell
rm -rf /usr/local/go && rm -rf /etc/profiles.d/go.sh
```

#### If as a sudoer:
```shell
sudo rm -rf /usr/local/go && sudo rm -rf /etc/profiles.d/go.sh
```

**Comment by obvionaoe** [comment]
Hey, thanks for all the info! I'll use that for now, would you mind pinging me here when you implement this?

Thanks and great work

**Comment by obvionaoe** [comment]
Another quick question while we're at it, does this role set the $GOPATH environment variable on the remote host? Would be fantastic if it did, as I'm building a role that would benefit from having that variable set. Thanks again

**Comment by darkwizard242** [comment]
@obvionaoe Sure, happy to notify you once I have updated the role the feature for removal of go.

Yes, it does add go to the `$PATH` however doesn't require to set a specific `$GOPATH` env var. I generally prefer to follow the [official documentation](https://golang.org/doc/install) and hence have added something similar.

This role adds a file `/etc/profile.d/go.sh`, which makes GO available systemwide to all users. The contents of the file are:
```shell
export PATH=$PATH:/usr/local/go/bin
```

Role has a template that intakes the value of role's ansible variable `go_bin_path` (defaults to `/usr/local`) to both extract Go and add the system-wide user profile but user can customize.

Basically allowing Go to be on the system's `$PATH`. By keeping it in the `/etc/profile.d` as a custom file - we are able to keep it dynamic with minimal changes for when we need to install or remove Go.

**Comment by obvionaoe** [comment]
How do you guarantee your file will be ran last in /etc/profile.d?

**Comment by darkwizard242** [comment]
hi @obvionaoe - I released version [1.1.2](https://github.com/darkwizard242/ansible-role-go/releases/tag/1.1.2) of the role today with changes that will facilitate to both install or uninstall go using the role. A boolean type variable has been introduced called `go_setup`. By default, the value of the variable is set to `true` to facilitate installation if a user has not otherwise set a different value such as `false` to override it.

For instance, if you want to install - then you just pass value for the variable **go_setup** as `true` in the playbook:
```yaml
- hosts: servers
  roles:
    - darkwizard242.go
  vars:
    go_version: 1.17.1
    go_setup: true
```

Should you wish to uninstall - you can pass `false` as the value for variable **go_setup** in the playbook:
```yaml
- hosts: servers
  roles:
    - darkwizard242.go
  vars:
    go_setup: false
```

**Comment by obvionaoe** [comment]
Thank you very much! I also read through the code and believe there's a typo in this line https://github.com/darkwizard242/ansible-role-go/blob/72b06873055f4ab832c55cbe3db25577cb8875ff/tasks/main.yml#L23 as you're importing the install_el.yml file twice instead of the uninstall_el.yml file!

Thanks again!

**Comment by darkwizard242** [comment]
Ahh good catch @obvionaoe - that was a typo on my end, thanks for cross-checking 👍🏼

**Comment by obvionaoe** [comment]
No worries, happy to help! :)
